From: Liu Bo <bo.li.liu@oracle.com>
Date: Wed, 28 Nov 2012 10:43:11 +0000
Patch-mainline: 3.8
Git-commit: 755ac67f83e515af55adbfe55134eb7d90839cdb
References: FATE#312888
Subject: [PATCH] Btrfs: skip adding an acl attribute if we don't have
 to

If the acl can be exactly represented in the traditional file
mode permission bits, we don't set another acl attribute.

Signed-off-by: Liu Bo <bo.li.liu@oracle.com>
Signed-off-by: Chris Mason <chris.mason@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/acl.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/btrfs/acl.c
+++ b/fs/btrfs/acl.c
@@ -123,6 +123,8 @@ static int btrfs_set_acl(struct btrfs_tr
 			ret = posix_acl_equiv_mode(acl, &mode);
 			if (ret < 0)
 				return ret;
+			if (ret == 0)
+				acl = NULL;
 			inode->i_mode = mode;
 		}
 		ret = 0;
